File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.cpp
class Solution {
public:
    int maximumElementAfterDecrementingAndRearranging(vector<int>& a ) {
        
        int n=a.size();
        sort(a.begin(),a.end());
        
        a[0]=1;
        for( int i=1;i<n;i++){
            if(abs(a[i]-a[i-1])<=1) continue;
            else{
                a[i]=a[i-1]+1;
            }
        }
        // for( int x:a) cout<<x<<" ";
        // cout<<"\n";
        
        return *max_element(a.begin(),a.end());
        
        
    }
};

File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.java
class Solution {
    public int maximumElementAfterDecrementingAndRearranging(int[] a) {
        
        int n=a.length;
        Arrays.sort(a);
    
        a[0]=1;
        int x=a[0];
        for( int i=1;i<n;i++){
            if(Math.abs(a[i]-a[i-1])>1) {
                a[i]=a[i-1]+1;
            }
            x=Math.max(x,a[i]);
        }
        return x;
        
    }
}